globals/prefixes.go

```diff
@@ -320,13 +322,26 @@ var resourceTypeToPrefixes map[resource.Type][]string = func() map[resource.Type
 	return ret
 }()
 
-// RegisterPrefixSubtype is called from various packages to register which
-// prefixes belong to their subtypes. This lets the subtypes stay in different
+// RegisterPrefixToResourceInfo is called from various packages to register
+// which prefixes belong to their them, what types those represent, and any
+// domain and subtype information. This lets the subtypes stay in different
 // packages (important for the reflection introspection we do) while not
 // creating import loops.
-func RegisterPrefixSubtype(prefix string, subtype Subtype) {
-	resInfo := prefixToResourceType[prefix]
+//
+// If called more than once for the same prefix, this function will panic.
+func RegisterPrefixToResourceInfo(prefix string, res resource.Type, domain string, subtype Subtype) {
+	resInfo, ok := prefixToResourceType[prefix]
+	if !ok && domain != "test" {
+		panic(fmt.Sprintf("prefix %q being registered to domain %q type %q subtype %q but did not already exist in map", prefix, domain, res.String(), subtype.String()))
+	}
+	if domain != "test" &&
+		(resInfo.Domain != "" && resInfo.Domain != domain ||
+			resInfo.Subtype != UnknownSubtype && resInfo.Subtype != subtype) {
```

Comment:
Note the check above. If we have something registered already and it's not part of an explicit test, we already panic above.

The reason I have this is because tons of tests panic without it. I can dig into it if we want, it's always targets, so there's probably a specific reason related to target's own registry.

Reply:
Ah -- I think I found it, I think it's a line I thought I'd removed already :-/

Reply:
Fixed